### ISSUE TITLE
Avoid ssa for Manifest CR when no update needed

### ISSUE THREAD

**Issue by ruanxin** [kyma-project/lifecycle-manager#1456]
### Description

At the moment, kyma controller will do ssa for all enabled modules, no matter if there is change or not, and in most cases, the related manifest CRs only need to be updated if there is update in the moduletemplate.

Solution:

Based on the research https://github.com/kyma-project/lifecycle-manager/issues/1481, we can use the cached manifest CR to avoid in cluster request to fetch the latest status

Kyma `status.modules[].version` tracking the current enabled manifest CR version, before doing ssa of manifest, we should compare it with the version of the current moduletemplate, only update manifest when the version get updated.

Include the check of comparing the `status.modules[].channel` as well.

https://github.com/kyma-project/lifecycle-manager/blob/bb2bf933d260730000b6e7f56b50fa0e5dfd0809/pkg/module/sync/runner.go#L121-L132

### Reasons

reduce requests to API server.

### Acceptance Criteria
- [x] Only do manifest CR update, by extending this [function](https://github.com/kyma-project/lifecycle-manager/blob/bb2bf933d260730000b6e7f56b50fa0e5dfd0809/pkg/module/sync/runner.go#L162), when: 
    - [x] There is new version for the moduletemplate 
    - [x] Or the channel has been switched 
    - [x] Or Manifest CR status.state has been changed compared to KymaCr.Status.Modules[x].state 
- [x] Cover scenario in tests.
- [x] Extend [this test](https://github.com/kyma-project/lifecycle-manager/blob/a0c49436f3d11d03c9a7556ec11c7c9f69d621d9/tests/e2e/module_upgrade_new_version_test.go#L12) to check Module Version  in `kyma.status` and ManifestCR gets updated in E2E

### Feature Testing

Unit Tests, check if possible for integration or E2E test

**Comment by nesmabadr** [comment]
This will be tackled in another issue because we are depending on this patch request to update the module status in kyma, so it needs further investigation. For now, we only added the version check in the kyma.status.Modules on upgrade in the E2E test

**Comment by nesmabadr** [comment]
Blocked by https://github.com/kyma-project/lifecycle-manager/issues/1481

**Comment by LeelaChacha** [comment]
Results on DEV:

~ 25% decrease in API server requests
no manifest reconciliation errors found
